backend/src/routes/api/notebooks/index.ts

```diff
@@ -49,6 +51,19 @@ module.exports = async (fastify: KubeFastifyInstance) => {
           ? false
           : await getNotebookStatus(fastify, namespace, name);
 
+        const notebookName = notebook.metadata.name;
```

Comment:
Notebook can be null.

Please also go update `getNotebook` with `Promise<Notebook | null>` as the return type.